### DISCUSSION TITLE
Hello. I did not see the "prefetch_line" function being called in the SMS.CC and SMS.h files. May I ask in which code segment the prefetch addresses generated by the SMS prefetcher are submitted to the processor for data prefetching

### DISCUSSION THREAD

**Discussion by szuhzw** [CMU-SAFARI/Hermes#29] [General]
@rahulbera

**Comment by rahulbera** [comment]
We initialize/operate SMS via `multi.l2c_pref` or `multi.llc_pref` prefetcher files (if you initialize SMS as L2 or LLC prefetcher, respectively). If you are initializing SMS as an L2 prefetcher (meaning, it's logically sitting beside L2 and observing all L1D miss requests), then [multi.l2c_pref:176](https://github.com/CMU-SAFARI/Hermes/blob/0cf5f8c2ee15358703bf34324dc1a9147b1a324a/prefetcher/multi.l2c_pref#L176) calls the `prefetch_line` function. If you are initializing SMS as an LLC prefetcher (meaning, it's logically sitting beside LLC and observing all L2C miss requests), then [multi.llc_pref:109](https://github.com/CMU-SAFARI/Hermes/blob/0cf5f8c2ee15358703bf34324dc1a9147b1a324a/prefetcher/multi.llc_pref#L109).

**Reply by szuhzw** [reply]
Thank you so much.
